Add new Elastic intro section (#2795)
elastic/docs-content: get-started/versioning-availability.md

description: Learn how Elastic handles versioning and feature availability in the docs. Find the product versions that are supported, how to read availability badges, and...
---

# Versioning and availability

> Learn how Elastic products are versioned, the lifecycle of features, and how to find the right documentation for your deployment type and product version. Find answers to common questions about the versioning and confidently navigate our continuously updated documentation.

Reviewer: Should we make this page a lot less about the docs and more about the product lifecycle?

If it's more about the docs, then we can merge it into the "how to use the documentation" page.

Learn how Elastic Docs handles versioning, feature availability, and how to find the right documentation for your deployment type and product version. Find answers to common questions about the Elastic Stack versioning and confidently navigate our continuously updated documentation.
Learn how Elastic products are versioned, the lifecycle of features, and how to find the right documentation for your deployment type and product version. Find answers to common questions about the versioning and confidently navigate our continuously updated documentation.

> ## Frequently asked questions

Reviewer 2: be cautious about removing this stuff. most of this was added to stop bleeding stemming from the migration.

> Each deployment type has a unique versioning strategy. For example, Serverless and ECH have releases that are distinguished by their date. Because Serverless is automatically updated, you always have the latest updates.

Reviewer: > Serverless and ECH have releases that are distinguished by their date.

Is that necessary to say? I think this adds more confusion than clarity.

We could rather say that: 
- Most of Elastic products share the same versioning pattern, based on the Elastic Stack: Elasticsearch, Kibana, ...
- Orchestrators such as ECE and ECK, and other tools such as code clients and SDKs, are versioned independently of the Elastic Stack, but the pattern follows the same principles (X.Y.Z)
- The Elastic Cloud console and Serverless projects are always automatically updated with the latest updates

Author: Edited!

## Understanding versioning

### Where can I find documentation for the latest version of the {{stack}}?
Each deployment type has a unique versioning strategy. For example, Serverless and ECH have releases that are distinguished by their date. Because Serverless is automatically updated, you always have the latest updates.

You’re in the right place! All documentation for Elastic Stack 9.0.0 and later is available at [elastic.co/docs](https://www.elastic.co/docs), including the latest {{version.stack| M.M}} version and any future versions in the 9.x series.

Need docs for an earlier version? Go to [elastic.co/guide](https://www.elastic.co/guide).

### Why doesn't Elastic have separate documentation for each version?

Starting with {{stack}} 9.0.0, Elastic no longer publishes separate documentation sets for each minor release. Instead, all changes in the 9.x series are included in a single, continuously updated documentation set.

This approach helps:
* Reduce duplicate pages
* Show the full history and context of a feature
* Simplify search and navigation

> ### How do I know content was added in a specific version?

Reviewer 2: there was a lot of concern that our badges were not understandable. are we ready to remove this guidance? did you check with the person who added it?

> | Deployment type | The environment where the feature is available, for example, {{stack}}, {{serverless-full}}, {{ece}}, {{eck}} |

Reviewer: ```suggestion
| Deployment type | The environment where the feature is available, for example, self-managed, {{serverless-full}}, {{ece}}, {{eck}} |
```
stack isn't a deployment type, it's what gets deployed

| Lifecycle state | The development or support status of the feature, for example, GA, Technical preview, Beta |
| Version | The specific version the lifecycle state applies to |

### Lifecycle states

| Lifecycle state | Description |

> | ----- | ----- |
> | Generally Available (GA) | Production-ready feature. When unspecified, GA is the default |
> | Beta | Feature is nearing general availability but not yet production-ready |
> | Technical preview | Feature is in early development stage |
> | Unavailable | Feature is not supported in this deployment type or version |

Reviewer: ```suggestion
| ----- | ----- |
| Technical preview | Feature is in early development stage |
| Beta | Feature is nearing general availability but not yet production-ready |
| Generally Available (GA) | Production-ready feature. When unspecified, GA is the default |
| Deprecated | Feature is still usable but is set to be removed or replaced in a future update |
| Removed | Feature can no longer be used |
| Unavailable | Feature is not supported in this deployment type or version |
```
Ordering + end of life states

> ### Examples of where availability can vary

Reviewer 2: we also do some of this explanation on the following surfaces and you can mine them for language / make sure they align: 

https://www.elastic.co/docs/deploy-manage/deploy#about-orchestration
https://www.elastic.co/docs/deploy-manage/deploy#versioning-and-compatibility
https://www.elastic.co/docs/deploy-manage/upgrade/plan-upgrade

>Because some products are released on different schedules and have different architecture or network constraints, it is possible that a feature isn't available for some deployment types, or becomes available at a different time for each deployment type.

would be careful with this because a lot of functionality is controlled by the stack version

| Beta | Feature is nearing general availability but not yet production-ready |
| Technical preview | Feature is in early development stage |
| Unavailable | Feature is not supported in this deployment type or version |

### Examples of where availability can vary

| Category | Example |
|-------|-------------|
| **Elastic Stack versions** | [Elastic Stack](the-stack.md) version 9.0.0 and later, including 9.1.0 |
| **Deployment types** | [Elastic Cloud Serverless](/deploy-manage/deploy/elastic-cloud/serverless.md), [Elastic Cloud Hosted](/deploy-manage/deploy/elastic-cloud/cloud-hosted.md), [Elastic Cloud Enterprise (ECE)](/deploy-manage/deploy/cloud-enterprise.md), [Elastic Cloud on Kubernetes (ECK)](/deploy-manage/deploy/cloud-on-k8s.md), and [Self-managed deployments](/deploy-manage/deploy/self-managed.md) |
| **Deployment versions** | [Elastic Cloud Enterprise (ECE)](/deploy-manage/deploy/cloud-enterprise.md) 4.0.0 and later, [Elastic Cloud on Kubernetes (ECK)](/deploy-manage/deploy/cloud-on-k8s.md) 3.0.0 and later |
| **Serverless project types** | {{es}}, {{observability}}, and {{elastic-sec}}

> ## Find docs for your product version

Reviewer 2: I had to add this based on an internal request. please don't remove it.

> | Deployment versions | [{{ece}}](/deploy-manage/deploy/cloud-enterprise.md) 4.0.0 and later, [{{eck}}](/deploy-manage/deploy/cloud-on-k8s.md) 3.0.0 and later |

Reviewer: ```suggestion
| Orchestrator versions | [{{ece}}](/deploy-manage/deploy/cloud-enterprise.md) 4.0.0 and later, [{{eck}}](/deploy-manage/deploy/cloud-on-k8s.md) 3.0.0 and later |
```
Using "deployment versions" could be misleading as in ECloud, ECE, and ECK, the deployment version is the stack version of the deployment.

> | Serverless project types | {{es}}, {{observability}}, and {{elastic-sec}} |

Reviewer: ```suggestion
| Serverless project types | {{es}}, Elastic {{observability}}, and {{elastic-sec}} |
```
We should rather fix (and align...) the variables but would need to make sure we're not creating anything weird before :D
